### DISCUSSION TITLE
paddle ocr cls推理崩溃

### DISCUSSION THREAD

**Discussion by poison857** [PaddlePaddle/PaddleOCR#13654] [Q&A]
封装了一个dll，调用ocr接口来识别
只想创建一个对象，初始化一次模型，于是在创建PPOCR对象时加了static，这样就可以除第一次时间慢，之后识别速度就快
但是多次调用接口会崩溃，次数不限
定位是在cls推理的时候崩溃，ocr_cls.cpp:具体代码段如下
![20240815154126](https://github.com/user-attachments/assets/4fd855e9-b432-4b3a-9b93-bf98e7c39a39)
现在
1.如果不用static，不会崩溃，但是每次推理都需要创建新的对象和初始化模型
2.不使用cls，也就不会崩溃，但是有角度的字符就识别不了了

请问一下大家 有办法解决这个问题吗

**Comment by GreatV** [comment]
估计是哪里内存没有释放吧

**Reply by poison857** [reply]
我看不到this->predictor_->Run()里的代码了
